common/src/main/java/whocraft/tardis_refined/common/tardis/control/ship/MonitorControl.java

```diff
@@ -16,10 +19,19 @@ public class MonitorControl extends Control {
     @Override
     public boolean onRightClick(TardisLevelOperator operator, ConsoleTheme theme, ControlEntity controlEntity, Player player) {
         if (!player.level().isClientSide()){
+            ItemStack hand = player.getMainHandItem();
+
+            // Temporary for testing purposes
+            if (hand.is(Items.EXPERIENCE_BOTTLE)) {
+                operator.getPilotingManager().setFuel(operator.getPilotingManager().getMaximumFuel());
+                player.sendSystemMessage(Component.literal("Fueled up!"));
+                return true;
+            }
+
             boolean isSyncingKey = false;
             if (PlayerUtil.isInMainHand(player, ItemRegistry.KEY.get())){
-                KeyItem key = (KeyItem)player.getMainHandItem().getItem();
-                if (key.interactMonitor(player.getMainHandItem(),player, controlEntity, player.getUsedItemHand()))
+                KeyItem key = (KeyItem)hand.getItem();
```

Comment:
no! bad duzo! PlayerUtil kinda sucks for this kinda thing
Instead use use pattern matching https://docs.oracle.com/en/java/javase/17/language/pattern-matching-instanceof-operator.html#GUID-843060B5-240C-4F47-A7B0-95C42E5B08A7

Reply:
this wasnt meeee but ill change it anyways